docs/development.rst

```diff
@@ -83,3 +84,59 @@ install the bleeding-edge package directly from the tskit github repository requ
 to specify ``subdirectory=python``:
 
 ``python3 -m pip install git+https://github.com/tskit-dev/tskit.git#subdirectory=python``
+
+***********************************************
+Best Practices for adding to the tskit codebase
```

Comment:
This is a bit long for a title (these get rendered in the left navbar). How about, "Development workflow" or "Feature lifecycle" or something?

Reply:
or "Best Practices for Development"